File: 13-02-2026/problem_3_pattern_match/buggy/pattern_match.c
// ===================================================
// PROBLEM 3: PATTERN MATCH COUNT - BUGGY C
// ===================================================
#include <stdio.h>
#include <string.h>

int main() {
    char t[200005], p[100005];
    fgets(t, 200005, stdin);
    fgets(p, 100005, stdin);
    
    int n = strlen(t);
    int m = strlen(p);
    int cnt = 0;
    
    for (int i = 0; i <= n - m; i++) {
        if (strncmp(t + i, p, m - 1) == 0) {
            cnt++;
        }
    }
    
    printf("%d\n", cnt);
    return 0;
}

File: 13-02-2026/problem_3_pattern_match/buggy/pattern_match.java
// ===================================================
// PROBLEM 3: PATTERN MATCH COUNT - BUGGY JAVA
// ===================================================
import java.io.*;

public class Solution {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String t = br.readLine();
        String p = br.readLine();
        int cnt = 0;
        
        for (int i = 0; i < t.length() - p.length(); i++) {
            if (t.substring(i, i + p.length()) != p) {
                cnt++;
            }
        }
        
        System.out.println(cnt);
    }
}
